File: 2021/June/Day 1/nextGreaterElement.py
# source : https://leetcode.com/problems/next-greater-element-i/submissions/
class Solution(object):
    def nextGreaterElement(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        out =[];

        for i in range(len(nums1)):
            for j in range(len(nums2)):
                if(nums1[i]==nums2[j]):
                    checkMat  = nums2[j:];
                    mins = 9999999999;
                    for elem in checkMat:
                        if(elem>nums1[i]):
                            mins = elem
                            out.append(elem);
                            break;
                    if(mins==9999999999):
                        out.append(-1);
        return out

File: 2023/January/Day 23/nextGreaterElement.cpp
// Source : https://leetcode.com/problems/next-greater-element-i/submissions/
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        vector<int> out;
        for(auto &i:nums1)
        {
            int f=-1;
            for(int j=0;j<size(nums2);j++)
            {
                if(nums2[j]==i)
                {
                    f=1;
                }
                if(f==1)
                {
                    if(nums2[j]>i)
                    {
                        f=2;
                        out.insert(out.end(),nums2[j]);
                        break;
                    }
                }
            }
            if(f!=2){out.insert(out.end(),-1);} 
        }
        return out;
    }
};